pages/blog/_posts/announcing-batch-keys.mdx

```diff
@@ -0,0 +1,87 @@
+---
+heading: "Announcing: Batch Keys"
+image: /assets/blog/announcing-batch-keys/featured-image.png
+showSubtitle: true
+subtitle: Batch keys allows developers to group work units by leveraging Inngest's efficient event-matching engine.
+date: 2024-07-25
+author: Charly Poly
+disableCTA: false
+---
+
+We are thrilled to release Batch Keys, a powerful addition to _Event Batching_. This new feature allows developers to group work units by leveraging [Inngest's efficient event-matching engine](https://www.inngest.com/blog/accidentally-quadratic-evaluating-trillions-of-event-matches-in-real-time), unlocking new advanced use cases such as Batch User Notifications or more efficient processing of large volumes of events.
+
+[*Event Batching*](https://www.inngest.com/docs/guides/batching), released a few months ago, has been widely adopted by developers and teams using Inngest to build highly scalable features such as ETL, notification systems, and third-party integrations. This feature enabled applications dealing with considerable throughput of events to group similar events in batches to create aggregates, reducing the number of requests to third-party APIs.
+
+![Without and with Event Batching](/assets/blog/announcing-batch-keys/batch-keys-1.jpg)
+
+Developers quickly started to apply *Event Batching* to product-oriented use cases, such as “Grouping similar notifications”, resulting in added complexity in the Functions' body to partition the group events based on event properties, for example, when sending post likes notifications:
+
+```tsx
+export const sendPostLikesNotifications = inngest.createFunction(
+  {
+    id: "send-post-likes-notification",
+    batchEvents: {
+      maxSize: 100, // let's group up to 100 post likes
+      timeout: "60s", // we want to group likes from the same 1min window
+    },
+  },
+  { event: "post.liked" },
+  async ({ events }) => {
+    // let's count the number of likes per post
+    const likes: Record<string, number> = events.reduce(
+      (acc, event) => {
+        if (!acc[event.data.post_id]) {
+          acc[event.data.post_id] = 1;
+        } else {
+          acc[event.data.post_id] += 1;
+        }
+        return acc;
+      },
+      {} as Record<string, number>,
+    );
+
+    for (let [postId, likesCount] of Object.entries(likes)) {
+	    await notifyPostLikesCount(postId, likesCount);
+    }
+  },
+);
+```
+
+The above implementation comes with multiple drawbacks:
+
+- **Fairness issues:** By batching by events, a post getting significantly more likes than others will benefit from a better grouping, impacting the end-user experience.
```

Comment:
```suggestion
- **Fairness issues:** By batching events, a post getting significantly more likes than others will benefit from a better grouping, impacting the end-user experience.
```

I think it's a typo?

Reply:
It is intentional, in comparison to Batching by event properties (with `keys`)

Reply:
gotcha! Do you think a clarification (grammatical or contextual) could work?

```suggestion
- **Fairness issues:** When batching by events, a post getting significantly more likes than others will benefit from a better grouping, impacting the end-user experience.
```

or:

```suggestion
- **Fairness issues:** By batching by events (and not by event properties), a post getting significantly more likes than others will benefit from a better grouping, impacting the end-user experience.
```

It might be a "me" thing so I'll file this as a nit :)